ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs

```diff
@@ -93,6 +92,27 @@ import           Ouroboros.Consensus.Storage.LedgerDB.InMemory
 type LedgerDB'       blk = LedgerDB       (ExtLedgerState blk)
 type AnnLedgerError' blk = AnnLedgerError (ExtLedgerState blk) blk
 
+{-------------------------------------------------------------------------------
+  Newtype wrappers for not mixing up old and new filesystems
+-------------------------------------------------------------------------------}
+
+-- | The directory containing old-style snapshots. These are snapshots taken
+-- from @ExtLedgerState blk ValuesMK@ in the same style as before the UTxO-HD
+-- integration
+newtype OldLedgerFS m = OldLedgerFS (SomeHasFS m)
+  deriving newtype (NoThunks)
+
+-- | The directory containing both the new-style snapshots and the UTxO-HD
+-- backend.
+--
+-- The new-style snapshots are serializations of @ExtLedgerState blk EmptyMK@ as
+-- carried in the @DbChangelog@.
+--
+-- INVARIANT: Only one of said snapshots should exist at any time TODO: this
```

Comment:
I don't know if we should mention this invariant here: there's nothing we can do at this level to enforce it. Maybe we want to place it in the function that takes the snapshots.